React/events.md

```diff
@@ -8,84 +8,119 @@
 
 ### Context
-In this lecture, we are going to expand upon the `NameDisplay` component that we created in the [state lecture](state.md). This was a great component that we created that was storing a name and a boolean in the state. We used the boolean in the render function to determine what to display.
+
+In this lecture, we are going to expand upon the `Student` component that we created in the [state lecture](state.md). This was a great component that we created that was storing a student's present status in the component. We used the boolean in the function to determine what to display.
 
 ### Adding More Meaning
+
 #### Toggle our Boolean
-The boolean value that we're storing in the state seems a little bit useless right now. We set it to `true` within the constructor and we don't ever change it! We will set up a button and handle the `onClick` event to affect the state within this component.
 
-Let's add a button that we will use to toggle the display of the name variable. We'll first add the button element within the `return` of our `render` function. Then we'll add the `onClick` attribute whose value will refer to an event handler function that we will create next.
+The boolean value that we're storing in the state seems a little bit useless right now. We set it to `true` as an initial value and we don't ever change it! We have a button and it doesn't do anything!  Next we will add code to change the state of `present` when the button is clicked on.
+
+### onClick Event
+
+Now we'll add the `onClick` attribute to the button whose value will refer to an event handler function that we will create next.
 
 ```javascript
-// NameDisplay.js
-  render() {
+// src/components/Student.js
+const Student = (props) => {
+
+  const [present, setPresent] = useState(props.present);
+  const [fullName, setFullName] = useState(props.fullName);
+
+  // Function to toggle present
+  const onButtonClick = () => setPresent(!present);
+
+    // Component functions always return JSX
     return (
-      ...
-      <button onClick={ this.onButtonClick }>Toggle Display</button>
+      <div>
+        <h3>{props.fullName}</h3>
+        <ul>
+          <li>Class: C13</li>
+          <li>Birthday: {props.birthday}</li>
+          <li>Email: {props.email}</li>
+        </ul>
+        <button onClick={onButtonClick}>
+          Mark {present ? 'Absent' : 'Present'}
+        </button>
+      </div>
     );
-  }
+}
 ```
 
+**Pause here** for a moment with the person sitting next to you. What is this code doing? Refer back to the [state lecture](state.md) if you can't remember what the `setPresent` function is supposed to do.
+
+**Note** the arrow function syntax for this event handler function. You could define the event handler function `onButtonClick` as a regular function, but making it an arrow function in this case is a bit more condense and keeps the `this` context to the current component.  
```

Comment:
```suggestion
**Note** the arrow function syntax for this event handler function. You could define the event handler function `onButtonClick` as a regular function, but making it an arrow function in this case is a bit more concise and keeps the `this` context to the current component.  
```

Reply:
Note: `this` is far less meaningful when using hooks instead of classes and may not be worth mentioning here.